From 1f4629f0104cb8760c0951912613b0b9c550a07f Mon Sep 17 00:00:00 2001
From: Wen He <wen.he_1@nxp.com>
Date: Fri, 19 Apr 2019 15:15:21 +0800
Subject: [PATCH 497/741] arm64: dts: ls1028a: Add properties for HDP
 Controller.

commit fd6c2a00068c94f01eb839b5f546e67281de8766 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

This patch enables the HDP controller driver on the LS1028A.

Signed-off-by: Alison Wang <aslion.wang@nxp.com>
Signed-off-by: Wen He <wen.he_1@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 .../arm64/boot/dts/freescale/fsl-ls1028a.dtsi | 25 +++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index 680cb6ebf650..9504c2a85efe 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -91,6 +91,13 @@
                 clock-output-names= "pclk";
         };
 
+        hdpclk: clock-hdpcore {
+                compatible = "fixed-clock";
+                #clock-cells = <0>;
+                clock-frequency = <162500000>;
+                clock-output-names= "hdpclk";
+        };
+
 	reboot {
 		compatible ="syscon-reboot";
 		regmap = <&rst>;
@@ -577,7 +584,25 @@
  
                 port {
                         dp0_out: endpoint {
+				remote-endpoint = <&dp1_out>;
+                        };
+                };
+        };
+
+        display1: hdp@f200000 {
+                compatible = "fsl,ls1028a-dp";
+                reg = <0x0 0xf1f0000 0x0 0xffff>,
+                    <0x0 0xf200000 0x0 0xfffff>;
+                interrupts = <0 221 IRQ_TYPE_LEVEL_HIGH>;
+                clocks = <&sysclk>, <&hdpclk>, <&dpclk>,
+                         <&dpclk>, <&dpclk>, <&pclk>, <&dpclk>;
+                clock-names = "clk_ipg", "clk_core", "clk_pxl",
+                              "clk_pxl_mux", "clk_pxl_link", "clk_apb",
+                              "clk_vif";
  
+                port {
+                        dp1_out: endpoint {
+                                remote-endpoint = <&dp0_out>;
                         };
                 };
         };
-- 
2.17.1

